modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -2718,6 +2736,25 @@ private RestoreLogicalState applyLogicalUpdates(
         for (DatabaseLifecycleListener lsnr : getDatabaseListeners(cctx.kernalContext()))
             lsnr.afterLogicalUpdatesApplied(this, restoreLogicalState);
 
+        List<IgniteInternalFuture<?>> clearFuts = new ArrayList<>();
+
+        partitionRecoveryClearing.forEach((key, value) -> {
+                    CacheGroupContext grp = ctx.cache().cacheGroup(key.getGroupId());
+
+                    GridDhtLocalPartition part = grp.topology().localPartition(key.getPartitionId());
```

Comment:
I think `GridDhtLocalPartition part = grp.topology().forceCreatePartition(key.getPartitionId);` is a bit safer.